miio/vacuumcontainers.py

```diff
@@ -175,6 +190,28 @@ def is_water_box_attached(self) -> bool:
         """Return True is water box is installed."""
         return "water_box_status" in self.data and self.data["water_box_status"] == 1
 
+    @property
+    def is_water_box_carriage_attached(self) -> Optional[bool]:
+        """Return True if water box carriage (mop) is installed, None if sensor not
+        present."""
+        if "water_box_carriage_status" in self.data:
+            return self.data["water_box_carriage_status"] == 1
+        else:
+            return None
+
+    @property
+    def is_water_shortage(self) -> Optional[bool]:
+        """Returns True if water is low in the tank, None if sensor not present."""
+        if "water_shortage_status" in self.data:
+            return self.data["water_shortage_status"] == 1
+        else:
+            return None
+
+    @property
+    def is_child_lock(self) -> bool:
+        """Returns True if the child lock is active."""
+        return self.data["lock_status"] == 1
```

Comment:
This also needs to check for if the key is given.

Reply:
Thanks to your comment i noticed that `lock_status` doesn't actually contain the status of the child lock, a call to get_child_lock_status is necessary. Will remove this method